src/lib/utils/pcbManualEditEventHandler.ts

```diff
@@ -0,0 +1,148 @@
+import type { ManualTraceHint } from "@tscircuit/layout"
+import { getManualTraceHintFromEvent } from "@tscircuit/layout"
+import type { EditEvent } from "@tscircuit/manual-edit-events"
+import {
+  type AnyCircuitElement,
+  type PcbComponent,
+  type SourceComponentBase,
+} from "circuit-json"
+
+export interface PCBPlacement {
+  selector: string
+  center: { x: number; y: number }
+  relative_to: "group_center"
+  _edit_event_id?: string
+}
+
+export interface ManualEditState {
+  pcb_placements: PCBPlacement[]
+  edit_events: EditEvent[]
+  manual_trace_hints: ManualTraceHint[]
+}
+
+export const createInitialManualEditState = (): ManualEditState => ({
+  pcb_placements: [],
+  edit_events: [],
+  manual_trace_hints: [],
+})
+
+export const handlePcbEditEvents = (
```

Comment:
`handle` implies effects, `apply` is a better prefix here.

```suggestion
export const applyEditEvents = (
```

Reply:
`apply` can imply mutations, so it's not perfect fwiw, but `handle` is broader and can imply "global effects" in a react context, e.g. binding to window